source/common/config/utility.cc

```diff
@@ -159,37 +159,64 @@ std::string Utility::resourceName(const ProtobufWkt::Any& resource) {
       fmt::format("Unknown type URL {} in DiscoveryResponse", resource.type_url()));
 }
 
-std::vector<Stats::TagExtractorPtr>
-Utility::createTagExtractors(const envoy::api::v2::Bootstrap& bootstrap) {
-  std::vector<Stats::TagExtractorPtr> tag_extractors;
-
-  // Ensure no tag names are repeated.
+void Utility::detectTagNameConflict(const envoy::api::v2::Bootstrap& bootstrap) {
   std::unordered_set<std::string> names;
-  auto add_tag = [&names, &tag_extractors](const std::string& name, const std::string& regex) {
+
+  if (!bootstrap.stats_config().has_use_all_default_tags() ||
+      bootstrap.stats_config().use_all_default_tags().value()) {
+    for (const std::pair<std::string, std::string>& default_tag :
+         TagNames::get().name_regex_pairs_) {
+      names.emplace(default_tag.first);
+    }
+  }
+
+  for (const envoy::api::v2::TagSpecifier& tag_specifier : bootstrap.stats_config().stats_tags()) {
+    const std::string name = tag_specifier.tag_name();
     if (!names.emplace(name).second) {
       throw EnvoyException(fmt::format("Tag name '{}' specified twice.", name));
     }
+  }
+}
 
-    tag_extractors.emplace_back(Stats::TagExtractorImpl::createTagExtractor(name, regex));
-  };
+std::vector<Stats::TagExtractorPtr>
+Utility::createTagExtractors(const envoy::api::v2::Bootstrap& bootstrap) {
+  std::vector<Stats::TagExtractorPtr> tag_extractors;
```

Comment:
It looks like you know how many items will go into `tag_extractors`. You can use this with reserve to avoid any costly reallocations:

`tag_extractors.reserve(default_tag_size + bootstrap_tags_size)`

Reply:
Nice, I didn't know that. I roughly estimate the size but should I get precise size? 
d3b5e7f